content-src/components/DiscoveryStreamComponents/TopSites/TopSites.jsx

```diff
@@ -4,14 +4,138 @@
 
 import { connect } from "react-redux";
 import { TopSites as OldTopSites } from "content-src/components/TopSites/TopSites";
+import { TOP_SITES_MAX_SITES_PER_ROW } from "common/Reducers.jsm";
 import React from "react";
 
 export class _TopSites extends React.PureComponent {
+  // Find a SPOC that doesn't already exist in User's TopSites
+  getFirstAvailableSpoc(topSites, data) {
+    const { spocs } = data;
+    if (!spocs || spocs.length === 0) {
+      return null;
+    }
+
+    const userTopSites = new Set(
+      topSites.map(topSite => topSite && topSite.url)
+    );
+
+    // We "clean urls" with http in TopSiteForm.jsx
+    // Spoc domains are in the format 'sponsorname.com'
+    return spocs.find(
+      spoc =>
+        !userTopSites.has(spoc.url) &&
+        !userTopSites.has(`http://${spoc.domain}`) &&
+        !userTopSites.has(`http://www.${spoc.domain}`)
```

Comment:
This failed for https://www.netflix.com/ with spoc.url as https://netflix.com returned from adzerk server
<img width="1059" alt="dupe-netflix" src="https://user-images.githubusercontent.com/597734/63560419-cbbcbe80-c50a-11e9-8ed9-e9fc476e0995.png">

Reply:
For left alignment, organic sites menu option edit is broken

Reply:
Should we just add two extra checks here for https:// and https://www, that should cover almost all the cases

Reply:
+1 considering its still an experiment and should cover bulk of the cases